lib/PushManager.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Created by noam.arad on 10/23/2017.
+ */
+
+const io = require('socket.io-client');
+const util = require('util');
+const BackendClient = require('./BackendClientFactory').getBackendClient();
+const logger = require('../common/logger').getLogger('PushManager');
+const kalturaTypes = require('./kaltura-client-lib/KalturaTypes');
+const playListGenerator = require('./playlistGenerator/PlaylistGenerator');
+const RecordingManager = require('./recording/RecordingManager');
+
+let instance = null;
+
+class PushManager {
+
+    constructor()
+    {
+        if (!instance) {
+            instance = this;
+            instance.registeredEntries = {};
+        }
+        return instance;
+    }
+
+    registerEntry(entryId, partnerId, templateName, callback)
+    {
+        const This = this;
```

Comment:
you don't need this if you use arrow function insteadof  function(queueKey, queueKeyHash